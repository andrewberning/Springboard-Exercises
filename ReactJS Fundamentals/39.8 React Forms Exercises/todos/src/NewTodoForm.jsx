import React, {useState} from "react";
import { v4 as uuid } from 'uuid';
import './NewTodoForm.css';


function NewTodoForm({ createTodo }) {
  const [task, setTask] = useState("");

  const handleChange = evt => {
    console.log(evt.target.value);
    setTask(evt.target.value);
  };

  const handleSubmit = evt => {
    evt.preventDefault();
    createTodo({ task, id: uuid() });
    setTask("");
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <div>
          <label htmlFor="task">Task:</label>
          <input 
            id="task"
            name="task"
            type="text"
            onChange={handleChange}
            value={task}
          />
        </div>
        <button>Add Todo</button>
      </form>
    </div>
  )
}

export default NewTodoForm
