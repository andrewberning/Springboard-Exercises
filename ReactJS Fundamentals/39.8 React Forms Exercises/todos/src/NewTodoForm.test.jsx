import { expect, it, vi } from 'vitest';
import { render, fireEvent } from "@testing-library/react"
import '@testing-library/jest-dom';
import NewTodoForm from './NewTodoForm';

it("renders without crashing", function() {
  render(<NewTodoForm />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("runs the create function on form submit", function() {
  const createMock = vi.fn();
  const { getByText} = render(<NewTodoForm createTodo={createMock} />);
  
  fireEvent.click(getByText("Add Todo"));
  expect(createMock).toHaveBeenCalled();
})