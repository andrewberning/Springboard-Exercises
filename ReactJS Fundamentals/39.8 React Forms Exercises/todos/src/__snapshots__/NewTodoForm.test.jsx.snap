// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="task"
        >
          Task:
        </label>
        <input
          id="task"
          name="task"
          type="text"
          value=""
        />
      </div>
      <button>
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
